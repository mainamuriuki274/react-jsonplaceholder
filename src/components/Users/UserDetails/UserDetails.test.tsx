import renderer from 'react-test-renderer';
import { User } from '../../../interfaces/detailedUserInterface';
import { UserDetails } from './index';

const sampleUser: User = {
  id: 1,
  name: 'Leanne Graham',
  username: 'Bret',
  email: 'Sincere@april.biz',
  address: {
    street: 'Kulas Light',
    suite: 'Apt. 556',
    city: 'Gwenborough',
    zipcode: '92998-3874',
  },
  phone: '1-770-736-8031 x56442',
};

describe('User Details Component', () => {
  it('it matches snapshot', () => {
    const tree = renderer.create(<UserDetails user={sampleUser} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

export {};
