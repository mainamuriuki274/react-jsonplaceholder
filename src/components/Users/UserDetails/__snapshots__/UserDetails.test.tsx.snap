// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Details Component it matches snapshot 1`] = `
<div
  className="user_details_container"
>
  <div>
    <h2>
      Personal Details
    </h2>
    <form
      className="edit_field_form"
      onSubmit={[Function]}
    >
      <div
        className="form_group"
      >
        <div
          className="form_input"
        >
          <label
            className="form-label"
            htmlFor="name"
          >
            Name
          </label>
          Â 
          <input
            className="form-control form-control-sm"
            id="name"
            name="name"
            onChange={[Function]}
            required={true}
            type="text"
            value="Leanne Graham"
          />
        </div>
      </div>
    </form>
    <p>
      <span
        className="label"
      >
        Username
      </span>
      <span
        className="value"
      >
        Bret
      </span>
    </p>
  </div>
  <div>
    <h2>
      Contact Information
    </h2>
    <p>
      <span
        className="label"
      >
        Email
      </span>
      <span
        className="value"
      >
        Sincere@april.biz
      </span>
    </p>
    <p>
      <span
        className="label"
      >
        Phonenumber
      </span>
      <span
        className="value"
      >
        1-770-736-8031 x56442
      </span>
    </p>
  </div>
  <div>
    <h2>
      Address Details
    </h2>
    <p>
      <span
        className="label"
      >
        Street
      </span>
      <span
        className="value"
      >
        Kulas Light
      </span>
    </p>
    <p>
      <span
        className="label"
      >
        Suite
      </span>
      <span
        className="value"
      >
        Apt. 556
      </span>
    </p>
    <p>
      <span
        className="label"
      >
        City
      </span>
      <span
        className="value"
      >
        Gwenborough
      </span>
    </p>
    <p>
      <span
        className="label"
      >
        Zip Code
      </span>
      <span
        className="value"
      >
        92998-3874
      </span>
    </p>
  </div>
</div>
`;
